libcloudforensics/prompts.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Classes for displaying prompts to a user and running their choices."""
+import abc
+from typing import List, Optional, Callable
+
+
+class PromptOption:
+  """Class representing an available option in a prompt.
+
+  Attributes:
+    text (str): The text to be displayed for the option.
+  """
+
+  def __init__(self, text: str, *functions: Callable[[], None]) -> None:
+    """Builds a PromptOption.
+
+    Args:
+      text (str): The text to be displayed for this prompt option.
+      functions (Callable[[], Any]): The underlying functions of this prompt
+        option to called upon execution
+    """
+    self._text = text
+    self._functions = functions
+
+  @property
+  def text(self) -> str:
+    """The text to be displayed for this prompt option."""
+    return self._text
+
+  def Execute(self) -> None:
+    """Executes the underlying functions of this prompt option."""
+    for function in self._functions:
+      function()
+
+class Prompt(abc.ABC):
+  """Class representing a prompt to the user.
+
+  Attributes:
+    execution_order (int): A number allowing to specify the priority of this
+      prompt, for use in other classes (see PromptSequence).
+  """
+
+  def __init__(self, execution_order: int) -> None:
+    """Builds a Prompt.
+
+    Args:
+      execution_order (int): A number allowing to specify the priority for this
+        prompt's execution, for use in other classes.
```

Comment:
add 4 spaces for hanging indents, here and everywhere else in the PR to make it consequent with the code base